tools/query_breakdown/src/main/java/com/google/bigquery/QueryBreakdown.java

```diff
@@ -0,0 +1,128 @@
+package com.google.bigquery;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+/**
+ * This class is where the main logic lives for the algorithm that this tool utilizes. It will
+ * also be in charge of outputting the results.
+ */
+public class QueryBreakdown {
+
+  // global fields that keeps track of the minimum unparseable component so far
+  private int minimumUnparseableComp = Integer.MAX_VALUE;
+  private Node solution;
+
+  // the generated tree
+  private Node root;
+  private static Parser parser;
+
+  /**
+   * This is the method that will run QueryBreakdown given an original query and output
+   * it to the specified output file, or if that is null, generate a new file to put the output in.
+   * The provided timeLimit will stop the tool from running over a certain time.
+   */
+  public static void run(String originalQuery, String outputFile, int errorLimit) {
+
+    // determines which parser to use
+    parser = new CalciteParser();
+
+    // uses the loop function to generate and traverse the tree of possible error recoveries
+    loop(originalQuery, errorLimit);
+
+    // write termination logic for output (tracing the node back, reconstructing path, output)
+    return;
```

Comment:
no need for this statement.

Reply:
I can still see this return statement. can you please remove it first?

Reply:
yikes, my bad--should be removed now